mantidproject/mantid
# Exposed most of the methods from ComponentInfo

**Description of work.**
Exposed `ComponentInfo` to workspaces so that it can be accessed in a similar fashion to `SpectrumInfo` and `DetectorInfo`.

Also exported most of the relevant methods to the Python side.
 
**Report to:** @OwenArnold . 

**To test:**
Nothing to test, just make sure that the unit tests all pass.


To test, just run ctest -C Debug -R ComponentInfoTest -V

Fixes #22923 . 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
